sympy/discrete/z_transforms.py

```diff
@@ -0,0 +1,1388 @@
+"""
+#------------------------------------------------------------------------#
+#                                                                        #
+#                         z_transform of f[n]                            #
+#                         z_transform_inverse of F[z]                    #
+#                                                                        #
+#------------------------------------------------------------------------#
+# http://blog.espol.edu.ec/telg1001/transformada-z-pares-fn-y-fz-con-sympy-python/
+# Edison Del Rosario edelros@espol.edu.ec
+# 2023/03/27
+"""
+### import for z_transform
+from sympy.core import diff, S #, pi, I
+from sympy.core.add import Add
+from sympy.core.mul import Mul
+from sympy.core.numbers import Rational
+from sympy.core.power import Pow
+from sympy.core.symbol import Wild, Symbol #,Dummy, symbols,
+from sympy.core.function import ( expand, WildFunction)
+                                #,Derivative,Function, Lambda)
+from sympy.simplify import powsimp
+from sympy.simplify.simplify import simplify #,nsimplify
+#from sympy.polys.polyroots import roots
+from sympy.polys.polytools import pdiv,real_roots,degree,LC,factor,Poly
+from sympy.polys.partfrac import apart
+from sympy.functions.elementary.trigonometric import cos, sin, acos, atan #cot,tan,
+from sympy.functions.elementary.miscellaneous import sqrt
+from sympy.functions.elementary.complexes import Abs#, (re, arg,  polar_lift,
+                                                  #periodic_argument)
+from sympy.functions.elementary.exponential import log #, exp, exp_polar
+from sympy.functions.combinatorial.factorials import factorial
+from sympy.concrete.summations import summation #,Sum
+
+from sympy.functions.special.delta_functions import DiracDelta, Heaviside
+#from sympy.logic.boolalg import to_cnf, conjuncts, disjuncts, Or, And
+from sympy import Float, preorder_traversal
+from sympy.utilities.misc import debug
+
+##########################################################################
+# Helpers / Utilities
+##########################################################################
+
+
+class z_transformError(NotImplementedError):
```

Comment:
I would name the error class 
```suggestion
class ZTransformError(NotImplementedError):
```

Reply:
greetings @sylee957 , thanks for your comment. 
'ZT' comes from 'ZT = z_transform' used at test files. 'z_transforms' is the name of function proposed, like the example at laplace transforms to simplify asserts. ¿name conventions to follow?
I appreciate your feedback

Reply:
The naming convention here is that class names do not have underscores, but each word is capitalized. We follow pep8 conventions, se here: https://peps.python.org/pep-0008/#class-names